### PR TITLE
Add cancellation support for acquire token interactive

### LINKED ISSUE #3225: [Bug] Windows Broker and Embedded Web View do check cancellation tokens
Version: MSAL.NET @ 4.42
Platform: Windows 11, `net5.0-windows10.0.17763.0`
Auth flow: Broker and Interactive Web auth (embedded web view only)

**Repro**
@bgavrilMS - we have an easy repro internally and a current bugfix in the auth CLI.

```csharp
private class MsalHttpClientFactoryAdaptor : IMsalHttpClientFactory
        {
            private HttpClient instance;

            public MsalHttpClientFactoryAdaptor(HttpClient instance)
            {
                if (instance == null)
                {
                    throw new ArgumentNullException(nameof(instance));
                }

                this.instance = instance;
            }

            public HttpClient GetHttpClient()
            {
                // MSAL calls this method each time it wants to use an HTTP client.
                // We ensure we only create a single instance to avoid socket exhaustion.
                return this.instance;
            }
        }
//---

ver pca = PublicClientApplicationBuilder
    .Create(this.clientId.ToString())
    .WithAuthority(this.Authority)
    .WithBroker();

var tokenSource = new CancellationStokenSource();
var timeout = TimeSpan.FromSeconds(15);

tokenSource.CancelAfter(timeout);

await pca.AcquireTokenInteractive(scopes)
    .ExecuteAsync(tokenSource.Token);

// Expect OperationCancelledException to be thrown - but the Web or broker UI hangs forever.
```

**Expected behavior**
An `OperationCancelledException` should be thrown by calling `token.ThrowIfCancellationRequested()`.

**Possible solution**
Internally we have a workaround to run await a timer task that checks if the task is done every 100ms and after the timeout, returns a `Task.FromResult<T>(null)`.

**Additional context / logs / screenshots / links to code**
We have debugged the un-cancellable call to this Task that runs with a custom Task Scheduler https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/96ec2757d1a18c48eb9b7abc02028a82bc601cc9/src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WebView2WebUi.cs#L74

### PR DESCRIPTION
**Pull request by MichelZ**
Fixes #3225 

**Changes proposed in this request**
- Add checking of CancellationToken to DisplayDialog (NetCore)
- Check cancellation Token and Close Dialog on Cancellation
- Add Integration Test to NetCore

**Testing**
Integration Test added to NetCoreWinFormsWAM, you can set Autocancel Seconds and authentication will be cancelled automatically without user interaction

**Performance impact**
None expected

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WinFormsPanelWithWebView2.cs [hunk 1]
```diff
@@ -65,23 +68,47 @@ internal class WinFormsPanelWithWebView2 : Form
...
         {
             DialogResult uiResult = DialogResult.None;
-            InvokeHandlingOwnerWindow(() => uiResult = ShowDialog(_ownerWindow));
 
+            // Setup a Task to listen for cancellation
+            // On Cancellation we need to Close the Dialog if it's open
+            // We also need to Stop the Task when the Dialog is finished.
+            var cts = new CancellationTokenSource();
+            
+            Task.Run(() =>
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WinFormsPanelWithWebView2.cs:92]: Not sure about this. How about using  `using (token.Register(() => token.ThrowIfCancellationRequested())` ?

File: src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WebView2WebUi.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ public WebView2WebUi(CoreUIParent parent, RequestContext requestContext)
             AuthorizationResult result = null;
             var sendAuthorizeRequest = new Action(() =>
             {
-                result = InvokeEmbeddedWebview(authorizationUri, redirectUri);
+                result = InvokeEmbeddedWebview(authorizationUri, redirectUri, cancellationToken);
```
[the remark this answers is not in the document]
Comment (MichelZ) [reply src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WebView2WebUi.cs:41]: Yes, i've seen that one and have done the same for the "old" browser now

File: src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WinFormsPanelWithWebView2.cs (hunk 1 above)
Comment (MichelZ) [reply src/client/Microsoft.Identity.Client/Platforms/Features/WebView2WebUi/WinFormsPanelWithWebView2.cs:92]: I'm not sure about the Throwing, will that close the browser form? But I can certainly close the window using this and then throw... will adapt

**Comment by bgavrilMS** [comment]
Closing in favour of https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3450 (so that the CI runs). If you have better suggestion on working with remote branches, happy to use

**Comment by MichelZ** [comment]
> Closing in favour of #3450 (so that the CI runs). If you have better suggestion on working with remote branches, happy to use

https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers
It seems you can configure Azure DevOps so that pipelines can be started manually by a team member using /azp run

**Pull request closed without merging**